### ISSUE TITLE
Allow d["a"][0].a

### ISSUE THREAD

**Issue by Happypig375** [fsharp/fslang-suggestions#1109]
# Allow dotting into successive applications of the short indexer syntax

I propose we allow
```fs
let d = Unchecked.defaultof<Map<string, Map<int, {| a : obj |}>>>
d["a"][0].a // This value is not a function and cannot be applied.
```

The existing way of approaching this problem in F# is `(d["a"][0]).a`

https://github.com/dotnet/fsharp/issues/12549#issuecomment-1004818851

> in F# 6 there is no rule to allow the alternative interpretation as a left-associative series of index lookups for this case. Thus for compatibility reasons the parentheses are required.

## Pros and Cons 

The advantages of making this adjustment to F# are
- Following the incentive of https://github.com/fsharp/fslang-design/blob/main/FSharp-6.0/FS-1110-index-syntax.md#motivation
  > This is to make learning F# easier (less to learn) and allow simpler migration of existing code from Python, C# and other languages.
- Being concise and not writing unnecessary dots

The disadvantage of making this adjustment to F# is losing compatibility.

## Extra information

Estimated cost (XS, S, M, L, XL, XXL): S

Related suggestions: https://github.com/dotnet/fsharp/issues/12549

## Affidavit (please submit!)

Please tick this by placing a cross in the box:
* [x] This is not a question (e.g. like one you might ask on [stackoverflow](http://stackoverflow.com)) and I have searched stackoverflow for discussions of this issue
* [x] I have [searched both open and closed suggestions on this site](http://github.com/fsharp/fslang-suggestions/issues) and believe this is not a duplicate
* [x] This is not something which has obviously "already been decided" in previous versions of F#.  If you're questioning a fundamental design decision that has obviously already been taken (e.g. "Make F# untyped") then please don't submit it.

Please tick all that apply:
* [ ] This is not a breaking change to the F# language design
* [x] I or my company would be willing to help implement and/or test this

## For Readers

If you would like to see this issue implemented, please click the :+1: emoji on this issue. These counts are used to generally order the suggestions by engagement.

**Comment by cartermp** [comment]
This certainly is in the category of "feels like a bug even though it's by design today" kind of things. I'd be in favor - less surprises.

**Comment by baronfel** [comment]
This pops up in the Slack/Discord with some regularity, and @KathleenDollard and I have hit it recently as well.

**Comment by dsyme** [comment]
Yes, I agree we should fix this. I'll mark as approved in principle.

**Comment by thomasd3** [comment]
I updated the original StackOverflow question to point to this thread in case anyone wants to comment (https://stackoverflow.com/questions/70500857/could-this-be-a-f-6-indexer-bug/70673467)
